File: src/main/java/de/spries/fleetcommander/service/core/GameAccessParams.java
package de.spries.fleetcommander.service.core;

public class GameAccessParams {

	private int gameId;
	private String authToken;

	public GameAccessParams(int gameId, String authToken) {
		this.gameId = gameId;
		this.authToken = authToken;
	}

	public int getGameId() {
		return gameId;
	}

	public String getAuthToken() {
		return authToken;
	}

	public String toJson() {
		return "{\"gameAuthToken\": \"" + authToken + "\", \"gameId\": \""
				+ gameId + "\"}";
	}
}

File: src/main/kotlin/de/spries/fleetcommander/web/dto/GameAccessParams.kt
package de.spries.fleetcommander.web.dto

import com.fasterxml.jackson.annotation.JsonIgnore

data class GameAccessParams(@get:JsonIgnore
                            val gamePlayer: GamePlayer,
                            @get:JsonIgnore
                            val authToken: String) {

    fun getGameId() = gamePlayer.gameId

    @JsonIgnore
    fun getPlayerId() = gamePlayer.playerId

    fun getFullAuthToken() = "${getPlayerId()}:$authToken"

}
